lib.rs

```diff
@@ -128,6 +128,7 @@ impl<'a> Drop for Sentinel<'a> {
             self.shared_data.active_count.fetch_sub(1, Ordering::SeqCst);
             if panicking() {
                 self.shared_data.panic_count.fetch_add(1, Ordering::SeqCst);
+                self.shared_data.no_work_notify_all();
```

Comment:
No, because a worker thread keeps on working on jobs for as long as possible and reducing the expensive create/cleanup cycle of threads.